File: week02/이정현/SWEA_8931_제로.py
T = int(input())

for test_case in range(1, T + 1):
    # 숫자의 수 입력
    n = int(input().strip())

    # 숫자를 입력 받을 배열
    stack = []

    # 숫자 수 만큼 값을 입력 받음
    for _ in range(n):
        # 스텍 내에 값이 있고 현재 입력 받는 값이 0이면 pop
        a = int(input().strip())
        if stack and a == 0:
            stack.pop()
        # 그렇지 않으면 리스트에 값을 추가
        else:
            stack.append(a)

    # sum을 이용해서 리스트 값을 더해 출력
    print(f"#{test_case} {sum(stack)}")

File: week02/박정수/SWEA_8931_제로.java
import java.util.*;

class Solution
{
	public static void main(String args[]) throws Exception
	{
	
		Scanner sc = new Scanner(System.in);
		int T;
		T=sc.nextInt();
	
		for(int test_case = 1; test_case <= T; test_case++)
		{
			int K = sc.nextInt();
            Stack<Integer> stack = new Stack<>();
            for(int i = 0; i < K; i++){
                int num = sc.nextInt();
                if(num == 0){
                    stack.pop();
                }else{
                    stack.push(num);
                }
            }

            int sum = 0;
            while(!stack.isEmpty()){
                sum += stack.pop();
            }
		}
	}
}